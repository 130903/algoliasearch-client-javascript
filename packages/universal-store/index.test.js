import createStore from './';

it('will error without namespace', () => {
  expect(() => createStore()).toThrowErrorMatchingSnapshot();
  expect(() => createStore('')).toThrowErrorMatchingSnapshot();
  expect(() => createStore(5)).toThrowErrorMatchingSnapshot();
});

it('has the correct methods', () => {
  expect(createStore('algolia-js-client')).toMatchSnapshot();
});

it('Detects sessionStorage availability', () => {
  const store = createStore('algolia-js-client');

  expect(store.supportsSessionStorage('client')).toBe(false);
});

it('sets then gets', () => {
  const store = createStore('algolia-js-client');

  const data = { someData: true };
  expect(store.set('key', data)).toMatchObject(data);

  expect(store.get('key')).toMatchObject(data);
});

describe('sessionStorage works', () => {
  beforeAll(() => {
    const state = {};
    global.sessionStorage = {
      getItem: jest.fn(key => state[key]),
      setItem: jest.fn((key, value) => {
        state[key] = value;
        return state[key];
      }),
    };
  });

  it('Detects sessionStorage availability', () => {
    const store = createStore('algolia-js-client');

    expect(store.supportsSessionStorage('client')).toBe(true);
  });

  it('sets then gets', () => {
    const store = createStore('algolia-js-client');

    const data = { someData: true };
    expect(store.set('key', data)).toMatchObject(data);

    expect(store.get('key')).toMatchObject(data);
  });

  it('sets to sessionStorage', () => {
    const store = createStore('algolia-js-client');

    const data = { weirdData: true };
    expect(store.set('otherKey', data)).toMatchObject(data);

    expect(
      JSON.parse(global.sessionStorage.getItem('algolia-js-client')).otherKey
    ).toMatchObject(data);
  });
});
