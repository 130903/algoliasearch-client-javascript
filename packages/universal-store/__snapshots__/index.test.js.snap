// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has the correct methods 1`] = `
Object {
  "get": [Function],
  "set": [Function],
  "supportsSessionStorage": [Function],
}
`;

exports[`will error without namespace 1`] = `"The namespace should be a string, received \\"undefined\\""`;

exports[`will error without namespace 2`] = `"The namespace should be a string, received \\"\\""`;

exports[`will error without namespace 3`] = `"The namespace should be a string, received \\"5\\""`;
